course/src/Isometry.cc

```diff
@@ -247,5 +284,104 @@ const Matrix3 Matrix3::kIdentity({1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.
 const Matrix3 Matrix3::kZero({0.0, 0.0, 0.0}, {0.0, 0.0, 0.0}, {0.0, 0.0, 0.0});
 const Matrix3 Matrix3::kOnes({1.0, 1.0, 1.0}, {1.0, 1.0, 1.0}, {1.0, 1.0, 1.0});
 
+Isometry::Isometry(const Matrix3& rotation) {
+    rotation_ = rotation;
+    translation_ = Vector3(0.0, 0.0, 0.0);
+}
+
+Isometry::Isometry(const Vector3& translation, const Matrix3& rotation) {
+    rotation_ = rotation;
+    translation_ = translation;
+}
+
+Isometry Isometry::compose(const Isometry& isometry) const {
+    Matrix3 rotation_result = matrix_multiplication(rotation_, isometry.rotation_);
+    Vector3 translation_result = (rotation_ * isometry.translation_) + translation_;
+    return Isometry(translation_result, rotation_result);
+}
+
+Isometry Isometry::inverse() const {
+    auto det = rotation_.det();
+    if (almost_equal(det, 0., 4)) {
+        throw std::domain_error("Cannot get the inverse of the matrix");
+    }
+    Matrix3 rot_result;
+    rot_result[0][0] = (rotation_[1][1] * rotation_[2][2]) - (rotation_[2][1] * rotation_[1][2]);
+    rot_result[0][1] = (rotation_[0][2] * rotation_[2][1]) - (rotation_[2][2] * rotation_[0][1]);
+    rot_result[0][2] = (rotation_[0][1] * rotation_[1][2]) - (rotation_[1][1] * rotation_[0][2]);
+
+    rot_result[1][0] = (rotation_[1][2] * rotation_[2][0]) - (rotation_[2][2] * rotation_[1][0]);
+    rot_result[1][1] = (rotation_[0][0] * rotation_[2][2]) - (rotation_[2][0] * rotation_[0][2]);
+    rot_result[1][2] = (rotation_[0][2] * rotation_[1][0]) - (rotation_[1][2] * rotation_[0][0]);
+
+    rot_result[2][0] = (rotation_[1][0] * rotation_[2][1]) - (rotation_[2][0] * rotation_[1][1]);
+    rot_result[2][1] = (rotation_[0][1] * rotation_[2][0]) - (rotation_[2][1] * rotation_[0][0]);
+    rot_result[2][2] = (rotation_[0][0] * rotation_[1][1]) - (rotation_[1][0] * rotation_[0][1]);
+
+    rot_result = rot_result / det;
+    auto vector_result = rot_result * translation_;
+    vector_result = vector_result * (-1);
+
+    return Isometry(vector_result, rot_result);
+}
+
+Matrix3 Isometry::rotation() const { return rotation_; }
+
+Vector3 Isometry::transform(Vector3 translation) const { return Vector3(rotation_ * translation + translation_); }
+
+Vector3 Isometry::translation() const { return translation_; }
+
+Isometry Isometry::FromTranslation(const std::initializer_list<double>& values) {
+    if (values.size() != 3) {
+        throw std::range_error("Not enough values in initializer list");
+    }
+    auto it = values.begin();
+    Vector3 vector;
```

Comment:
You already have a Vector3 constructor that works from initializer lists, right?